### PR TITLE
Fixed #8298: added ZStream.readFile support for ScalaNative

### LINKED ISSUE #8298: Add ZStream.readFile support for ScalaNative

### PR DESCRIPTION
**Pull request by melsonic**
Fixes #8298 
/claim #8298

### CODE REVIEW

**Review by adamgfraser — CHANGES_REQUESTED** [review]
Can you please add tests for this?

**Comment by melsonic** [comment]
> Can you please add tests for this?

sure

File: streams/shared/src/main/scala/zio/stream/ZStream.scala [hunk 1]
```diff
@@ -6064,4 +6065,15 @@ object ZStream extends ZStreamPlatformSpecificConstructors {
...
       (cl.zipWith(cr.take(cl.size))(f), Right(cr.drop(cl.size)))
+
+  def readFile(path: => Path)(implicit trace: Trace, d: DummyImplicit): ZIO[Any, IOException, String] =
+    readFile(path.toString)
+
+  def readFile(name: => String)(implicit trace: Trace): ZIO[Any, IOException, String] =
+    ZIO.acquireReleaseWith(ZIO.attemptBlockingIO(scala.io.Source.fromFile(name)))(s =>
+      ZIO.attemptBlocking(s.close()).orDie
+    ) { s =>
+      ZIO.attemptBlockingIO(s.mkString)
```
Comment (jdegoes) [inline streams/shared/src/main/scala/zio/stream/ZStream.scala:6076]: Creating one string from `Source#mkString` is going to be very inefficient.

**Comment by adamgfraser** [comment]
@melsonic This was already implemented in #8364. Sorry for the confusion!

**Pull request closed without merging**